### ISSUE TITLE
[FEATURE]: Install PKGBUILD's from local paths

### ISSUE THREAD

**Issue by fries1234** [crystal-linux/amethyst#87]
### Is your feature request related to a problem? Please describe.

Like if you have a local copy of a PKGBUILD, you could use Amethyst to install them. Thinking about how to implement something like this, maybe we could uh copy the files to a cache folder or something?

### Describe the solution you'd like

Something like `ame ins --path .` (like `cargo install --path`).

### Describe alternatives you've considered

I don't really know any alternate solutions of how you could implement this into amethyst. I guess you could use `makepkg -si`.

**Comment by fries1234** [comment]
@mrshmllow what do you think?

**Comment by SomethingGeneric** [comment]
This similar to #73 and I totally agree with the need for both c:

**Comment by axtloss** [comment]
I don't see why this should be implemented into amethyst when the user could just run `makepkg`

**Comment by mrshmllow** [comment]
they have differing behaviours from my experimenting last night

**Comment by axtloss** [comment]
then that's a bug that should be reported, not a new feature

**Comment by Trivernis** [comment]
The advantage would be that ame is also able to handle aur dependencies while makepkg is not

**Comment by fries1234** [comment]
yeah that would be an advantage

**Comment by axtloss** [comment]
> The advantage would be that ame is also able to handle aur dependencies while makepkg is not

that's true, although how many would actually use that feature? it may not be worth the effort to implement and maintain it

**Comment by fries1234** [comment]
im pretty sure some people might use it. ngl i prob would use it as resolving dependencies for local aur packages can be annoying af.

**Comment by axtloss** [comment]
But why have locally cloned aur packages when you can just use amethyst to directly install it? amethyst is an aur helper, not a makepkg replacement

**Comment by fries1234** [comment]
well there could be pkgbuilds that arent published to the aur or that are out of date on the aur

**Comment by axtloss** [comment]
well that's that that makepkg is for

**Comment by fries1234** [comment]
but it doesnt install aur deps. if your pkgbuild has aur deps, it will just tell you to install those manually.

**Comment by axtloss** [comment]
is there any issues with manually installing dependencies? 
I still don't believe that many would actually use the feature since it's very rare to find a pkgbuild that's not in the aur

**Comment by Trivernis** [comment]
The thing is that this would be quite challenging to implement as we'd need to parse the PKGBUILD ourselves and I'm not sure if it would be worth the effort

**Comment by mrshmllow** [comment]
it is hard to test how helpers would handle the pkgbuild

**Comment by axtloss** [comment]
> it is hard to test how helpers would handle the pkgbuild 

why would it be needed to test how helpers handle pkgbuilds? they all would be handled the same way as makepkg. If an aur helper doesn't, then it's a bug that should be reported

**Comment by mrshmllow** [comment]
that is how you find bugs

**Comment by not-my-segfault** [comment]
> I don't see why this should be implemented into amethyst when the user could just run `makepkg`

`makepkg` cannot resolve AUR dependencies

**Comment by SomethingGeneric** [comment]
> The thing is that this would be quite challenging to implement as we'd need to parse the PKGBUILD ourselves and I'm not sure if it would be worth the effort

What am I missing that prevents you from using the same approach as an AUR package? Aren't you still parsing the PKGBUILD to get dependencies normally?

And if you do the rewrite for this, then a git spec shouldn't be hard either.

**Comment by not-my-segfault** [comment]
> > The thing is that this would be quite challenging to implement as we'd need to parse the PKGBUILD ourselves and I'm not sure if it would be worth the effort
> 
> What am I missing that prevents you from using the same approach as an AUR package? Aren't you still parsing the PKGBUILD to get dependencies normally?
> 
> And if you do the rewrite for this, then a git spec shouldn't be hard either.

We parse dependencies through the AUR's RPC, rather than the `PKGBUILD` directly, which is generally [what is recommended](https://wiki.archlinux.org/title/AUR_helpers#Legend) as attempting to parse any `depends` directly by `source`-ing the `PKGBUILD` could allow for arbitrary code execution in the case of a malicious `PKGBUILD`

**Comment by SomethingGeneric** [comment]
> > > The thing is that this would be quite challenging to implement as we'd need to parse the PKGBUILD ourselves and I'm not sure if it would be worth the effort
> > 
> > 
> > What am I missing that prevents you from using the same approach as an AUR package? Aren't you still parsing the PKGBUILD to get dependencies normally?
> > And if you do the rewrite for this, then a git spec shouldn't be hard either.
> 
> We parse dependencies through the AUR's RPC, rather than the `PKGBUILD` directly, which is generally [what is recommended](https://wiki.archlinux.org/title/AUR_helpers#Legend) as attempting to parse any `depends` directly by `source`-ing the `PKGBUILD` could allow for arbitrary code execution in the case of a malicious `PKGBUILD`

Perhaps then you could have an unsafe confirm flag? If it's a local path or git repo it's more likely the user looked over it, after
